src/main/java/io/cdap/plugin/gcp/gcs/StorageClient.java

```diff
@@ -329,6 +330,30 @@ private Blob move() {
       LOG.debug("Successfully deleted {}.", toPath(sourceBlob.getBlobId()));
       return moved;
     }
+  }
+
+  /**
+   * Creates a done file marker (empty file) in the given GCS path.
+   */
+  public void createFileMarker(GCSPath path) {
+    Bucket bucket = null;
+    String bucketName = path.getBucket();
+    String objectName = path.getName();
+    try {
+      bucket = storage.get(bucketName);
+    } catch (StorageException e) {
+
+      throw new RuntimeException(
+        String.format("Unable to access bucket %s. ", bucketName)
+          + "Ensure you entered the correct bucket path.", e);
+    }
+    if (bucket == null) {
+      throw new IllegalArgumentException(
+        String.format("Bucket '%s' does not exist.", bucketName));
+    }
 
+    BlobId blobId = BlobId.of(bucketName, objectName);
+    BlobInfo blobInfo = BlobInfo.newBuilder(blobId).build();
+    storage.create(blobInfo, "".getBytes(StandardCharsets.UTF_8));
```

Comment:
what happens if the path is already existing? In that case, we should not fail this call.

Reply:
Also does storage client return any exceptions?

Reply:
If an exact path to marker file exist, this doesn't fail.

Reply:
Do we need to catch any storage exceptions? The javadoc of `storage.create` should have that information